src/server/plugins/elasticsearch/lib/upgrade_config.js

```diff
@@ -3,16 +3,32 @@ var isUpgradeable = require('./is_upgradeable');
 var _ = require('lodash');
 var format = require('util').format;
 module.exports = function (server) {
+
+  var MAX_INTEGER = Math.pow(2, 53) - 1;
+
   var client = server.plugins.elasticsearch.client;
   var config = server.config();
   return function (response) {
     var newConfig = {};
-    // Check to see if there are any doc. If not then we can assume
-    // nothing needs to be done
-    if (response.hits.hits.length === 0) return Promise.resolve();
+
+    // Check to see if there are any doc. If not then we set the build number and id
+    if (response.hits.hits.length === 0) {
+      var isProduction = process.env.NODE_ENV === 'production';
+      return client.create({
+          index: config.get('kibana.index'),
+          type: 'config',
+          body: {
+            buildNum: isProduction ? config.get('kibana.buildNum') : MAX_INTEGER
+          },
+          id: isProduction ? config.get('kibana.package.version') : '@@version'
+        });
+    }
 
     // if we already have a the current version in the index then we need to stop
-    if (_.find(response.hits.hits, { _id: config.get('kibana.package.version') })) {
+    if (_.find(response.hits.hits, function currentVersion(hit) {
```

Comment:
@jbudz will you update this to [use descriptive-conditions](https://github.com/elastic/kibana/blob/master/STYLEGUIDE.md#use-descriptive-conditions) please?